src/main/java/graphql/nadel/hooks/ServiceExecutionHooks.java

```diff
@@ -32,15 +31,15 @@ default NewVariableValue visitArgumentValueInQuery(HooksVisitArgumentValueEnviro
         return null;
     }
 
-    default Optional<GraphQLError> isFieldAllowed(Field field, GraphQLFieldDefinition fieldDefinitionOverall, Object userSuppliedContext) {
-        return Optional.empty();
+
+    default CompletableFuture<Optional<GraphQLError>> isFieldAllowed(Field field, GraphQLFieldDefinition fieldDefinitionOverall, Object userSuppliedContext) {
```

Comment:
While we're here can we add some documentation, here I'll even write it

```java
/**
 * Called to determine whether a field should be omitted from the query to the underlying service.
 * When a field is omitted, a GraphQL error is inserted into the overall response in lieu of it.
 *
 * @param field                  the field to omit or not
 * @param fieldDefinitionOverall the field's definition
 * @param userSuppliedContext    the context supplied to Nadel in {@link graphql.nadel.NadelExecutionInput}
 * @return an error if the field should be omitted, empty optional otherwise
 */
```

Reply:
Next level: make it a suggestions so that I can apply it with one click: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request :)